WiseBuy/View Controllers/DealsViewController.m

```diff
@@ -30,28 +31,48 @@ -(void)viewDidLoad {
 }
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:YES];
-    NSLog(@"%@", self.barcode);
-//    [APIManager fetchDealsFromUPCDatabase:@"53039031"];
 
-    if (![DatabaseManager checkIfItemAlreadyExist:@"53039031"]) {
-        [APIManager fetchDealsFromEbayAPI:@"53039031"];
+    if (![DatabaseManager checkIfItemAlreadyExist:self.barcode]) {
+        [APIManager fetchDealsFromEbayAPI:self.barcode];
         [APIManager fetchDealsFromUPCDatabase:self.barcode];
         [APIManager fetchDealsFromSearchUPCAPI:self.barcode];
     }
+    JGProgressHUD *HUD = [JGProgressHUD progressHUDWithStyle:JGProgressHUDStyleLight];
+    HUD.textLabel.text = @"Waiting for deals to be displayed";
+    [HUD showInView:self.view];
     
-    [DatabaseManager fetchItem:@"53039031" viewController:self withCompletion:^(NSArray * _Nonnull deals, NSError * _Nonnull error) {
+    [self setLoadingState:YES viewController:self];
+    
+    [DatabaseManager fetchItem:self.barcode viewController:self withCompletion:^(NSArray * _Nonnull deals, NSError * _Nonnull error) {
```

Comment:
nit: When you have long functions like this, putting each param on a separate line makes it much easier to read